akka-actor-typed/src/main/scala/akka/actor/typed/internal/LoggerClass.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal
+
+import akka.annotation.InternalApi
+import akka.util.OptionVal
+
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object LoggerClass {
+
+  // just to get access to the class context
+  private final class TrickySecurityManager extends SecurityManager {
+    def getClassStack: Array[Class[_]] = getClassContext
+  }
+
+  /**
+   * Try to extract a logger class from the call stack, if not possible the provided default is used
+   */
+  def getLoggerClass(default: Class[_]): Class[_] = {
```

Comment:
perhaps rename to something that doesn't look as innocent as `getLoggerClass`, e.g. `findLoggerClass`